In [43]:
import pandas as pd
import numpy as np

In [35]:
# Read uncleaned disney_movies.csv file
df_2 = pd.read_csv(r'C:\Users\ashwi\Documents\Data Analytics\QA\Data\Cleaned\Python\df_2.csv', index_col=False, sep=',')

In [36]:
df_2

movie_name Running time (min) Release date  \
0               Pinocchio (1940film)              01:28   1940-02-07   
1                Fantasia (1940film)              02:06   1940-11-13   
2    The Reluctant Dragon (1941film)              01:14   1941-06-27   
3                              Dumbo              01:04   1941-10-23   
4                              Bambi              01:10   1942-08-09   
..                               ...                ...          ...   
366                   Chang Can Dunk              01:49   2023-03-10   
367                 Peter Pan  Wendy              01:49   2023-04-28   
368                    Crater (film)              01:45   2023-05-12   
369               Worlds Best (film)              01:41   2023-06-23   
370         Dashing Through the Snow              01:30   2023-11-17   

      Box office  
0    164000000.0  
1     76400000.0  
2       960000.0  
3      1300000.0  
4    267399999.0  
..           ...  
366          NaN  
367          NaN  
368          NaN  
369          NaN  
370          NaN  

[371 rows x 4 columns]

In [37]:
df_2.columns

Index(['movie_name', 'Running time (min)', 'Release date', 'Box office'], dtype='object')

In [40]:
# Define a function to process the movie names
def process_movie_name(name):
    import re
    # Regular expression to match sequences of 4 numbers followed by alpha characters
    pattern = r'(\d{4})([a-zA-Z]+)'
    # Check if the name contains parentheses
    if '(' in name and ')' in name:
        # If it does, apply the regular expression to the part inside the parentheses
        name = re.sub(pattern, r'\1 \2', name)
    return name

# Apply the function to the movie_name column
df_2['movie_name'] = df_2['movie_name'].apply(process_movie_name)

# Display the updated DataFrame
df_2

movie_name Running time (min) Release date  \
0               Pinocchio (1940 film)              01:28   1940-02-07   
1                Fantasia (1940 film)              02:06   1940-11-13   
2    The Reluctant Dragon (1941 film)              01:14   1941-06-27   
3                               Dumbo              01:04   1941-10-23   
4                               Bambi              01:10   1942-08-09   
..                                ...                ...          ...   
366                    Chang Can Dunk              01:49   2023-03-10   
367                  Peter Pan  Wendy              01:49   2023-04-28   
368                     Crater (film)              01:45   2023-05-12   
369                Worlds Best (film)              01:41   2023-06-23   
370          Dashing Through the Snow              01:30   2023-11-17   

      Box office  
0    164000000.0  
1     76400000.0  
2       960000.0  
3      1300000.0  
4    267399999.0  
..           ...  
366          NaN  
367          NaN  
368          NaN  
369          NaN  
370          NaN  

[371 rows x 4 columns]

In [41]:
df_2.to_csv('df_3.csv', index=False, sep=',', encoding='utf-8')

In [42]:
# Read uncleaned disney_movies.csv file
df_3 = pd.read_csv(r'C:\Users\ashwi\Documents\Data Analytics\QA\Data\Cleaned\Python\df_3.csv', index_col=False, sep=',')

# df_3

## Data Exploration

In [49]:
df_3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371 entries, 0 to 370
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   movie_name          371 non-null    object 
 1   Running time (min)  369 non-null    object 
 2   Release date        371 non-null    object 
 3   Box office          269 non-null    float64
dtypes: float64(1), object(3)
memory usage: 11.7+ KB


In [50]:
df_3.describe()

Box office
count  2.690000e+02
mean   1.935762e+09
std    3.048122e+10
min    1.600000e-01
25%    5.100000e+06
50%    2.290000e+07
75%    7.220000e+07
max    5.000003e+11

In [52]:
df_3.isnull().sum()

movie_name              0
Running time (min)      2
Release date            0
Box office            102
dtype: int64

Box office contains 102 null values

## Data Cleaning

### Handling Missing Values

In [ ]:
# df_3.drop(na)

In [ ]:
# df_3.fillna()

### Remnove Duplicates

## Quality Assurance

### Visualization libraries

#### Matplotlib

In [57]:
import matplotlib.pyplot as plt

#### Seaborn

#### Plotly

#### Z-Score

#### IQR

## Statistical Analysis

### Descriptive Statistics

In [58]:
df_3.describe()

Box office
count  2.690000e+02
mean   1.935762e+09
std    3.048122e+10
min    1.600000e-01
25%    5.100000e+06
50%    2.290000e+07
75%    7.220000e+07
max    5.000003e+11

#### Correlation analysis

In [ ]:
df_3.corr()

#### Hypothesis testing

In [ ]:
scipy.stats

In [ ]:
t-tests

In [ ]:
ANOVA

#### Regression analysis

In [ ]:
statsmodels

In [ ]:
scikit-learn